### PR TITLE
Added functionality for manyDeposits change in the escrow

### PR DESCRIPTION
**Pull request by carlosvdr**
Updated Escrow contract and added changes for the subgraph to work accordingly.

- Added a many deposits ID (this value will be filled with the txn ID) looking through this value you can find all Deposits related to this txn

### CODE REVIEW

File: tests/local_contract_calls.py [hunk 1]
```diff
@@ -85,6 +95,7 @@ class ReceiptAggregateVoucher(EIP712Message):
     print("Approving escrow contracts")
     erc20.functions.approve(ESCROW_ADDRESS, 10000).transact({"from": GATEWAY})
     erc20.functions.approve(ESCROW_ADDRESS, 10000).transact({"from": SIGNER})
+    erc20.functions.approve(ESCROW_ADDRESS, 10000).transact({"from": SIGNER})
```
Comment (aasseman) [inline tests/local_contract_calls.py:98]: Is it normal that it's the same line twice?

File: tests/local_contract_calls.py [hunk 2]
```diff
@@ -95,8 +106,18 @@ class ReceiptAggregateVoucher(EIP712Message):
...
+
+    # Convert the hash to the Ethereum specific signature format
+    eth_signed_message = encode_defunct(hexstr=message_hash.hex())
+
+    # Sign the message with the private key
+    signature = w3.eth.account.sign_message(eth_signed_message, private_key=ALLOCATIONID_PK)
+    arbitraryBytes32 = os.urandom(32)
     print("Allocate receiver with allocationid")
-    mockStaking.functions.allocate(ALLOCATION_ID, RECEIVER).transact({"from": GATEWAY})
+    mockStaking.functions.allocate(arbitraryBytes32, 1000, ALLOCATION_ID, arbitraryBytes32, RECEIVER).transact({"from": RECEIVER})
```
Comment (aasseman) [inline tests/local_contract_calls.py:120]: This is OK, but ideally should be in another PR, since it's not about supporting `manyDeposits`.

File: src/mappings/escrow.ts [hunk 3]
```diff
@@ -49,7 +49,7 @@ export function handleCancelThaw(event: CancelThaw): void {
 }
 
 export function handleDeposit(event: Deposit): void {
-    let transaction = new Transaction(event.transaction.hash.toHexString())
+    let transaction = new Transaction(event.transaction.hash.toHexString() + '-' + event.logIndex.toString())
```
Comment (aasseman) [inline src/mappings/escrow.ts:52]: Not sure if it makes sense, but could this be moved in a function such as `newTransaction(event)` to make it cleaner and less error-prone?

**Review by aasseman — CHANGES_REQUESTED** [review]

File: tests/local_contract_calls.py (hunk 1 above)
Comment (carlosvdr) [reply tests/local_contract_calls.py:98]: yeah no, added it for a test forgot to remove it

File: src/mappings/escrow.ts (hunk 3 above)
Comment (carlosvdr) [reply src/mappings/escrow.ts:35]: Not sure if its the best approach on that, Ill ask Juan about this what he thinks might be best in this regard.

Comment (carlosvdr) [reply src/mappings/escrow.ts:35]: Will add it into a new PR

File: tests/local_contract_calls.py (hunk 2 above)
Comment (aasseman) [reply tests/local_contract_calls.py:119]: Nah, we'll keep it here, but it'd be better for next times.

**Review by aasseman — APPROVED** [review]
:ship
